src/it/MSHADE-313/verify.bsh

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import java.io.*;
+import java.util.jar.*;
+
+String[] wanted =
```

Comment:
I'm missing the `META-INF/services/DependencyServiceInterface`, which is quite critical to be verified, right?

Reply:
As minijar only filters `*.class` files, and as [MSHADE-313](#) does not change that strategy, I did not see any good reason to check for the existence this resource file in the outcome (why should it be gone just by adding more class files?). MSHADE-313 does not provide a complete solution for service handling, it just fixes one single bug: prevent deleting needed classes. But as you want to have that check, I just added it right now in 440d6eb0340c7812a8130b3f960739f254e54ea0.